### PR TITLE
Enhancement/location var type

### PR DESCRIPTION
**Pull request by sky3d**
Support for new location type { lat, lng } is introduced

Required object with following fields: `lat`, `lng` in case-sensitive

Usage example:
```
def get_location():
    return {'lat': 48.858093, 'lng': 2.294694}

variables = {
        'Eiffel Tower': {
            'type': 'location',
            'bind': get_location
        }
}
```

### CODE REVIEW

File: cloud4rpi/utils.py [hunk 1]
```diff
@@ -73,6 +73,13 @@ def to_string(value):
         return str(value)
 
 
+def to_location(value):
+    if isinstance(value, dict):
+        return {x: value[x] for x in ('lat', 'lng')}
```
Comment (Himura2la) [inline cloud4rpi/utils.py:78]: It's not clear why make a dict from the same dict. What's wrong with another keys if we don't use them? Maybe so?
```
if not (isinstance(value, dict) and 'lat' in value and 'lng' in value):
    raise Exception()
return value
```

File: test/test_device.py [hunk 2]
```diff
@@ -609,3 +623,44 @@ def testStringAsBool(self):
...
         self.api.publish_data.assert_called_with({'Status': 'true'})
+
+    def testLocation(self):
+        location = {'lat': 37.89, 'lng': 75.43}
+        self.device.declare({'Pos': {'type': 'location'}})
+        self.device.publish_data({'Pos': location})
+        self.api.publish_data.assert_called_with({'Pos': location})
+
+    def testLocation_Filtering(self):
+        obj = {'some': 'foo', 'lng': 75.43, 'lat': 37.89, 'other': 42}
```
Comment (Himura2la) [inline test/test_device.py:634]: I don't think it's required to filter the dict

**Review by Himura2la — CHANGES_REQUESTED** [review]

File: cloud4rpi/utils.py (hunk 1 above)
Comment (sky3d) [reply cloud4rpi/utils.py:78]: We don't use payload fields except 'lat' and 'lng' on a server side. It's not need to transfer anything else.

File: test/test_device.py (hunk 2 above)
Comment (sky3d) [reply test/test_device.py:634]: See above.

**Review by const314 — APPROVED** [review]

**Review by Himura2la — APPROVED** [review]

**Comment by Himura2la** [comment]
docs https://github.com/cloud4rpi/docs/pull/27

**Pull request merged**